### PR TITLE
An Overview of Fuzzy Logic System

### LINKED ISSUE #885: An Overview of Fuzzy Logic System
### Brief Summary: 
Fuzzy logic is a computing technique that is based on the degree of truth. It is applied in various industries such as finance, manufacturing and automobile. This article provides readers with a basic understanding of a fuzzy logic system. 

### Key Takeaways: 
1. What is a fuzzy logic?
2. Fuzzy logic architecture
3. Using Fozzy logic (algorithm)
4. Fuzzy logic membership function
5. Applications of fuzzy logic
6. Conclusion

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

### PR DESCRIPTION
**Pull request by mbaabuones**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #885 .

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

File: articles/an-overview-of-fuzzy-logic-system/index.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Fuzzy logic systems are used in automobile and domestic applications to control actions and processes. This article provides an overview of this logic and how it is implemented. It also highlights its real-life applications and explains why it is used.
+
+### What is Fuzzy Logic?
+Fuzzy means uncertain, indefinite, vague, or unclear. Fuzzy logic is a computing technique that is based on the degree of truth. A fuzzy logic system uses the input's degree of truth and [linguistic variables](https://www.igi-global.com/dictionary/granular-computing/17211) to produce a certain output. The state of this input determines the nature of the output.
+
+This technique is different from [Boolean logic](https://www.bbc.co.uk/bitesize/guides/zc4bb9q/revision/1), which uses only two categories (true or false). In Boolean logic, two categories (0 and 1) are used to describe objects. For example, the degree of temperature in water served in glass may be High (1) or Low (0). In fuzzy logic, the water is described using more categories, but within the two categories mentioned earlier. In this case, the water may be very cold, very warm, or warm.
```
Comment (lalith1403) [inline articles/an-overview-of-fuzzy-logic-system/index.md]: Boolean  logic -> *boolean logic*. Don't capitalize, use italics or bold.

File: articles/an-overview-of-fuzzy-logic-system/index.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Fuzzy logic systems are used in automobile and domestic applications to control actions and processes. This article provides an overview of this logic and how it is implemented. It also highlights its real-life applications and explains why it is used.
+
+### What is Fuzzy Logic?
+Fuzzy means uncertain, indefinite, vague, or unclear. Fuzzy logic is a computing technique that is based on the degree of truth. A fuzzy logic system uses the input's degree of truth and [linguistic variables](https://www.igi-global.com/dictionary/granular-computing/17211) produce a certain output. The state of this input determines the nature of the output.
+
+This technique is different from [Boolean logic](https://www.bbc.co.uk/bitesize/guides/zc4bb9q/revision/1), which uses only two categories (true or false). In Boolean logic, two categories (0 and 1) are used to describe objects. For example, the temperature in water served in glass may be High (1) or Low (0). The water is described using more categories in fuzzy logic, but within the two categories mentioned earlier. In this case, the water may be very cold, very warm, or warm.
```
Comment (lalith1403) [inline articles/an-overview-of-fuzzy-logic-system/index.md:8]: High or Low -> italics.

File: articles/an-overview-of-fuzzy-logic-system/index.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### What is Fuzzy Logic?
+Fuzzy means uncertain, indefinite, vague, or unclear. Fuzzy logic is a computing technique that is based on the degree of truth. A fuzzy logic system uses the input's degree of truth and [linguistic variables](https://www.igi-global.com/dictionary/granular-computing/17211) produce a certain output. The state of this input determines the nature of the output.
+
+This technique is different from [Boolean logic](https://www.bbc.co.uk/bitesize/guides/zc4bb9q/revision/1), which uses only two categories (true or false). In Boolean logic, two categories (0 and 1) are used to describe objects. For example, the temperature in water served in glass may be High (1) or Low (0). The water is described using more categories in fuzzy logic, but within the two categories mentioned earlier. In this case, the water may be very cold, very warm, or warm.
+
+Let's take another simple example. Suppose we have a question that we need to answer. In Boolean logic, the answer would be either yes or no. In fuzzy logic, the answer may be between these two categories. Some of the probable answers in this logic include possibly yes, possibly no, or certainly no.
```
Comment (lalith1403) [inline articles/an-overview-of-fuzzy-logic-system/index.md:10]: *boolean logic*. Change such occurrences throughout the article.

File: articles/an-overview-of-fuzzy-logic-system/index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+The fuzzy logic architecture consists of the following components:
+
+* **Rule Base:** This contains the rules and membership functions that regulate or control decision-making in the fuzzy logic system. It also contains the IF-THEN conditions used for conditional programming and controlling the system. 
+* **Fuzzifier:** This component transforms raw inputs into fuzzy sets. The fuzzy sets proceed to the control system, where they undergo further processing. 
+* **Inference Engine:** This is a tool that establishes the ideal rules for a specific input. It then applies these rules to the input data to generate a fuzzy output. 
+* **Defuzzifier:** This component transforms the fuzzy sets into an explicit output (in the form of crisp inputs). Defuzzification is the final stage of a fuzzy logic system. 
+  
+### Fuzzy Logic Membership Function
+A membership function is a graphical representation of a fuzzy set. It shows how values ranging between 0 and 1 are mapped to inputs. Inputs are generally represented as Universe (U). The membership function is in the form μ A:X → [0,1], where A is a fuzzy set and X is the Universe.
```
Comment (lalith1403) [inline articles/an-overview-of-fuzzy-logic-system/index.md:37]: Represent the equation using latex. 

$$ \mu (A):X ->[0,1] $$ . Change it accordingly. also, what is the relation between mu and A?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Well written article.

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/an-overview-of-fuzzy-logic-system/index.md (hunk 1 above)
Comment (mbaabuones) [reply articles/an-overview-of-fuzzy-logic-system/index.md]: Done.

File: articles/an-overview-of-fuzzy-logic-system/index.md (hunk 2 above)
Comment (mbaabuones) [reply articles/an-overview-of-fuzzy-logic-system/index.md]: Done.

File: articles/an-overview-of-fuzzy-logic-system/index.md (hunk 3 above)
Comment (mbaabuones) [reply articles/an-overview-of-fuzzy-logic-system/index.md]: Done.

File: articles/an-overview-of-fuzzy-logic-system/index.md (hunk 4 above)
Comment (mbaabuones) [reply articles/an-overview-of-fuzzy-logic-system/index.md]: $$\mu_{A} simply refers to the membership function of set A. I have represented the equation using latex. The equation looks good in the preview section of Visual Studio Code. However, the appearance is different here in Github. Please notify me if further changes are needed.

**Comment by mbaabuones** [comment]
@lalith1403 I have finished working on the changes.

**Review by lalith1403 — APPROVED** [review]

**Pull request merged**